src/Aspire.Hosting.Azure/AzureResourceExtensions.cs

```diff
@@ -22,4 +23,21 @@ public static IResourceBuilder<T> PublishAsConnectionString<T>(this IResourceBui
         ParameterResourceBuilderExtensions.ConfigureConnectionStringManifestPublisher((IResourceBuilder<IResourceWithConnectionString>)builder);
         return builder;
     }
+
+    /// <summary>
+    /// Gets the Bicep identifier for the Azure resource.
+    /// </summary>
+    /// <param name="resource">The Azure resource.</param>
+    /// <returns>A valid Bicep identifier.</returns>
+    public static string GetBicepIdentifier(this IAzureResource resource) =>
+        NormalizeBicepIdentifier(resource.Name);
+
+    /// <summary>
+    /// Normalizes the given identifier name to make it a valid Bicep identifier name.
+    /// </summary>
+    /// <param name="identifierName">The identifier name to normalize.</param>
+    /// <returns>A valid Bicep identifier.</returns>
+    // TODO: This API should be deleted when we get the API from Azure.Provisioning
+    public static string NormalizeBicepIdentifier(string identifierName) =>
```

Comment:
@tg-msft is going to add an API for this in Azure.Provisioning. I'm just putting this API here for now until that API comes in.

There are problems with using shared source files and internal classes. The unit test references them all and fails to compile since it has IVT to all the libraries and so it sees multiple copies. This was easiest for now.

Reply:
The fact that we can’t do this in the provisioning context is kinda depressing

Reply:
> The fact that we can’t do this in the provisioning context is kinda depressing

How/when would you expect that to happen? On insertion? All at the end?

There are places where we reference these names, or find by these names, where the names need to line up.